bugtool/cmd/root.go

```diff
@@ -56,14 +75,17 @@ func init() {
 func runTool() {
 	// Search for a Cilium pod and if one is found prefix all of the
 	// commands to use the pod(s).
-	k8sPods, _ = getCiliumPods(k8sNamespace, k8sLabel)
-	if len(k8sPods) == 0 {
-		common.RequireRootPrivilege("bugtool")
+	var err error
+	k8sPods, err = getCiliumPods(k8sNamespace, k8sLabel)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "....Failed to retrieve cilium logs from kubernetes: %s\n", err)
```

Comment:
Does this always happen even if you run in an environment without kubernetes?

Reply:
The root requirement is not for Kubernetes but for the case where we detect no pods and would like to run all the commands on the host. In that case, we need root possibly for some commands and copying the state directory or we see something like

```
Could not copy state directory open /var/run/cilium/state: permission denied
```

Reply:
What I meant to ask about is this: if there's no kubernetes available then presumably `getCiliumPods()` will always fail, so users of the tool would always see this error message. However, that error message would make no sense to report in a non-kubernetes environment.